src/components/EditTaskModal.tsx

```diff
@@ -0,0 +1,140 @@
+import "antd/dist/antd.css";
+import { DatePicker } from "antd";
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { Button, Input, Modal, Select } from "semantic-ui-react";
+import { Column, Task } from "./Board";
+import { Moment } from "moment";
+import { ColorItem } from "./ColorItem";
+import { updateTaskAction } from "../redux/actions/editTaskActions";
+import moment from "moment";
+
+interface EditTaskModalInterface {
+  onClose: () => void;
+  taskToEdit: Task;
+  columns: Column[];
+}
+
+export default function EditTaskModal(
+  props: EditTaskModalInterface
+): JSX.Element {
+  const dispatch = useDispatch();
+
+  const taskToEdit = props.taskToEdit;
+
+  const [inputValue, setEditInputValue] = useState(taskToEdit?.title);
+  const [dateDeadline, setDateDeadeline] = useState(taskToEdit?.date);
+  const [taskColumnId, setTaskColumnId] = useState(taskToEdit?.columnId);
+  const [cardColor, setCardColor] = useState(taskToEdit?.color);
+
+  const colorOptions = [
+    {
+      value: "yellow",
+      text: <ColorItem color="#CEFF00" />,
+    },
+    {
+      value: "blue",
+      text: <ColorItem color="#0048BA" />,
+    },
+    {
+      value: "green",
+      text: <ColorItem color="#00FF00" />,
+    },
+    {
+      value: "red",
+      text: <ColorItem color="#FF0800" />,
+    },
+    {
+      value: "teal",
+      text: <ColorItem color="#008080" />,
+    },
+    {
+      value: "olive",
+      text: <ColorItem color="#808000" />,
+    },
+  ];
+  const dateFormat = "DD-MM-YYYY";
+
+  const handleChangeCardColor = (
+    _event: React.SyntheticEvent<HTMLElement, Event>,
+    data: { value?: any }
+  ) => {
+    setCardColor(data.value);
+  };
+
+  const handleChangeTaskTitile = (
+    _event: React.ChangeEvent<HTMLInputElement>,
+    data: { value?: any }
+  ) => {
+    setEditInputValue(data.value);
+  };
+
+  const handleDateSelect = (date: Moment) => {
+    setDateDeadeline(date.format(dateFormat).toString());
+  };
+
+  const handleSaveModal = () => {
+    dispatch(
+      updateTaskAction(
+        taskToEdit.id,
+        inputValue,
+        cardColor,
+        dateDeadline,
+        taskColumnId
+      )
```

Comment:
keep in mind important rule: 
If a function takes a lot of arguments (more than 3) - make it 1 object argument.

In your current code it's easy to mess up the order

Reply:
probably fixed